### ISSUE TITLE
Docker failed to start in 4.1.1 Linux Container

### ISSUE THREAD

**Issue by soroshsabz** [docker/for-win#12270]
ITNOA

  - [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 5BCDC324-A8CA-44F2-9CDA-DE2A62E609EB/20211014120311

### Actual behavior

When I switch to Linux container I see below exception

```
System.InvalidOperationException:
Failed to deploy distro docker-desktop to C:\Users\ssoroosh\AppData\Local\Docker\wsl\distro: exit code: -1
 stdout: The network was not found.


 stderr: 
   at Docker.ApiServices.WSL2.WslShortLivedCommandResult.LogAndThrowIfUnexpectedExitCode(String prefix, ILogger log, Int32 expectedExitCode) in C:\workspaces\PR-16599\src\github.com\docker\pinata\win\src\Docker.ApiServices\WSL2\WslCommand.cs:line 146
   at Docker.Engines.WSL2.WSL2Provisioning.<DeployDistroAsync>d__17.MoveNext() in C:\workspaces\PR-16599\src\github.com\docker\pinata\win\src\Docker.Engines\WSL2\WSL2Provisioning.cs:line 168
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.Engines.WSL2.WSL2Provisioning.<ProvisionAsync>d__8.MoveNext() in C:\workspaces\PR-16599\src\github.com\docker\pinata\win\src\Docker.Engines\WSL2\WSL2Provisioning.cs:line 77
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.Engines.WSL2.LinuxWSL2Engine.<DoStartAsync>d__29.MoveNext() in C:\workspaces\PR-16599\src\github.com\docker\pinata\win\src\Docker.Engines\WSL2\LinuxWSL2Engine.cs:line 194
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.ApiServices.StateMachines.TaskExtensions.<WrapAsyncInCancellationException>d__0.MoveNext() in C:\workspaces\PR-16599\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\TaskExtensions.cs:line 29
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.ApiServices.StateMachines.StartTransition.<DoRunAsync>d__5.MoveNext() in C:\workspaces\PR-16599\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\StartTransition.cs:line 67
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Docker.ApiServices.StateMachines.StartTransition.<DoRunAsync>d__5.MoveNext() in C:\workspaces\PR-16599\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\StartTransition.cs:line 92

```

### Expected behavior

I expect docker switch to linux container correctly, without any error.

### Information

Please, help us understand the problem.  For instance:
  - Is it reproducible? Yes
  - Is the problem new? No, from version 4.0.0 related to #12066
  - Did the problem appear with an update?  No (but this problem is occur when I install new [Windows Cumulative Update  2021/09](https://support.microsoft.com/en-us/topic/september-1-2021-kb5005101-os-builds-19041-1202-19042-1202-and-19043-1202-preview-82a50f27-a56f-4212-96ce-1554e8058dc1)

  - Windows Version: Windows 10 Version 21H1 (OS Build 19043.1237)
  - Windows Feature Experience Pack: 120.2212.3530.0
  - Docker Desktop Version: 4.1.1
  - WSL2 or Hyper-V backend?   I use Windows Container (but WSL 2 based engine is enable in setting)
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: No

My docker settings like below

```
{
  "registry-mirrors": [],
  "insecure-registries": [],
  "debug": false,
  "experimental": false
}
```

### Steps to reproduce the behavior

  1. Try to run Docker Desktop
  2. Docker Desktop try to start Docker Engine
  3. Docker Engine failed to start after couple of minutes
  4. I cannot do anything :(

**Comment by soroshsabz** [comment]
```
Please investigate the following 3 issues:

1 : are the LinuxKit services running?
    Failed with: failed to ping VM diagnosticsd with error: Get "http://ipc/ping": open \\.\pipe\dockerDiagnosticd: The system cannot find the file specified.
The Docker engine runs inside a Linux VM as a service. Therefore the services must have started.

2 : are the backend processes running?
    Failed with: 4 errors occurred:
        * vpnkit-bridge.exe is not running
        * vpnkit.exe is not running
        * com.docker.proxy.exe is not running
        * com.docker.wsl-distro-proxy.exe is not running


Not all of the backend processes are running.

3 : is the VM networking working?
    Failed with: network checks failed: Post "http://ipc/check-network-connectivity": open \\.\pipe\dockerDiagnosticd: The system cannot find the file specified.
VM seems to have a network connectivity issue. Please check your host firewall and anti-virus settings in case they are blocking the VM.
```

How to resolve these issue?

**Comment by psantos10** [comment]
I also keep having the same issue related here: https://github.com/docker/for-win/issues/12229

It started when I upgraded to 4.1.0. To solve the issue, the only solution was downgrade to 4.0.1. 

Now, with 4.1.1, I upgraded, and at the time, it was working. But after restarting the PC the problem appears again. 

I cleanup everything. Deleted all docker related files and directory, and installed docker again. Again, was working but after restarting the PC, the problem still there.

![image](https://user-images.githubusercontent.com/374424/137473005-1fea6281-9a45-4023-8cb2-273aa701b1c8.png)

Diagnostics ID: 4C300C44-6CBC-497C-A809-ED0E1440597E/20211015102406
